src/main/java/org/jenkinsci/plugins/gitclient/GitUsernamePasswordBind.java

```diff
@@ -0,0 +1,156 @@
+            final UnbindableDir unbindTempDir = UnbindableDir.create(filePath);
+            setRunEnviornmentVariables(filePath, taskListener);
+            GenerateGitScript gitScript = new GenerateGitScript(credentials.getUsername(),
+                                                credentials.getPassword().getPlainText(), credentials.getId());
+            FilePath gitTempFile = gitScript.write(credentials, unbindTempDir.getDirPath());
+            credMap.put("GIT_ASKPASS", gitTempFile.getRemote());
+            return new MultiEnvironment(credMap, unbindTempDir.getUnbinder());
+        } else {
+            return new MultiEnvironment(credMap);
+        }
+    }
+
+    @Override
+    public Set<String> variables() {
+        Set<String> keys = new LinkedHashSet<>();
+        keys.add(USERNAMEKEY);
+        keys.add(PASSWORDKEY);
+        return keys;
+    }
+
+
+    @Override
+    public void setKeyBindings(@Nonnull StandardCredentials credentials) {
+        credMap.put(USERNAMEKEY,((StandardUsernamePasswordCredentials) credentials).getUsername());
+        credMap.put(PASSWORDKEY,((StandardUsernamePasswordCredentials) credentials).getPassword().getPlainText());
+    }
+
+    @Override
+    public void setRunEnviornmentVariables(@Nonnull FilePath filePath, @Nonnull TaskListener listener) throws IOException, InterruptedException {
+        if(!Functions.isWindows() && ((CliGitAPIImpl) getGitClientInstance(listener)).
+                                            isAtLeastVersion(2,3,0,0))
+        {
+            credMap.put("GIT_TERMINAL_PROMPT","false");
+        }else {
+            credMap.put("GCM_INTERACTIVE","false");
+        }
+    }
+
+    @Override
+    public GitClient getGitClientInstance(TaskListener listener) throws IOException, InterruptedException {
+        Git gitInstance = new Git(listener,new EnvVars()).using(GIT_TOOL_NAME);
+        return gitInstance.getClient();
+    }
+
+    protected static final class GenerateGitScript extends AbstractOnDiskBinding<StandardUsernamePasswordCredentials> {
+
+        private String userVariable = null;
+        private String passVariable = null;
+
+        protected GenerateGitScript(String gitUsername, String gitPassword, String credentialId) {
+            super(gitUsername + ":" + gitPassword, credentialId);
+            this.userVariable = gitUsername;
+            this.passVariable = gitPassword;
+        }
+        @Override
+        protected FilePath write(StandardUsernamePasswordCredentials credentials, FilePath workspace)
+                throws IOException, InterruptedException {
+            FilePath gitEcho;
+            if (!Functions.isWindows()) {
+                gitEcho  = workspace.createTempFile("auth", ".sh");
+                gitEcho.write("#!/bin/sh\n" +
+                        "case $1 in\n" +
+                        "        Username*) echo " + this.userVariable + "\n" +
```

Comment:
Should `username` and `password` perhaps be all caps or all lowercase?
```suggestion
                        "        username*) echo " + this.userVariable + "\n" +
```

Reply:
The `Username` and `Password` are the part of the script, so this shouldn't  be changed, here the script is performing a switch case. Looking for the `Username` keyword on the terminal prompt.